common/metrics/src/json_metrics.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+use std::process::Command;
+
+// Use to expose non-numeric metrics
+pub fn get_json_metrics() -> HashMap<String, String> {
+    let mut json_metrics: HashMap<String, String> = HashMap::new();
+    json_metrics = add_revision_hash(json_metrics);
+    serde_json::to_string(&json_metrics).unwrap();
+    json_metrics
+}
+
+fn add_revision_hash(mut json_metrics: HashMap<String, String>) -> HashMap<String, String> {
+    let output = Command::new("git")
+        .args(&["rev-parse", "--short", "HEAD"])
+        .output()
+        .unwrap();
+    let git_hash = &String::from_utf8(output.stdout).unwrap();
+    json_metrics.insert("revision".to_string(), git_hash.trim().to_string());
+    json_metrics
+}
```

Comment:
This probably isn't what you want. This would require that when running a validator that it must be run inside of a git repository, checked out to the revision that it was built on.